pkg/clustermanager/cluster_manager.go

```diff
@@ -790,6 +800,43 @@ func (c *ClusterManager) InstallMachineHealthChecks(ctx context.Context, cluster
 	return nil
 }
 
+// UpdateMachineHealthChecks updates machine health checks for a given eksa cluster.
+func (c *ClusterManager) UpdateMachineHealthChecks(ctx context.Context, clusterSpec *cluster.Spec, workloadCluster *types.Cluster) error {
+	mhcCP, err := c.clusterClient.GetMachineHealthCheck(ctx, workloadCluster, clusterapi.ControlPlaneMachineHealthCheckName(clusterSpec.Cluster))
+	if err != nil {
+		return err
+	}
+
+	if mhcCP != nil && machineHealthCheckDiffExists(clusterSpec.Cluster.Spec.MachineHealthCheck, mhcCP) {
+		err = c.InstallMachineHealthChecks(ctx, clusterSpec, workloadCluster)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func machineHealthCheckDiffExists(eksaMachineHealthCheck *v1alpha1.MachineHealthCheck, capiMachineHealthCheck *clusterv1.MachineHealthCheck) bool {
+	if eksaMachineHealthCheck == nil && capiMachineHealthCheck == nil {
+		return false
+	}
+
+	if eksaMachineHealthCheck == nil || capiMachineHealthCheck == nil {
+		return true
+	}
+
+	if len(eksaMachineHealthCheck.NodeStartupTimeout) != len(capiMachineHealthCheck.Spec.NodeStartupTimeout.Duration.String()) {
```

Comment:
What if it is 20m00s vs 20m0s? Is that going to be counted as a difference because of this check?